### DISCUSSION TITLE
Jøkul hovedmenykomponent

### DISCUSSION THREAD

**Discussion by Steinop** [fremtind/jokul#2283] [Annet]
Komponentene vi bruker i Jøkul (eller som visuelt fremstår som komponenter) bør være tilgjengelige til felles bruk både i Figma og kode. 
I sammenheng med blå hovereffekt bør vi fjerne blå markering i active state og kun vise blå på hover og focus her også.
Komponenten må samtidig ha en dark mode versjon i Figmabiblioteket og i kode.

**Comment by Steinop** [comment]
Figmakomponent basert på hovedmenyen som ligger i dagens versjon av portalen. Denne har hover og focus states i tillegg til minimert versjon på scroll. På grunn av Regular font i default state og Bold i active trenger vi både regular og bold versjon av Hover state. Lar det seg gjøre @piofinn ? Ser at det allerede fungerer slik på NavLink.

Slik ser menyen ut uten blå markering i active state:
https://vimeo.com/447802624/c193a85a09

**Comment by Steinop** [comment]
Hovedmenykomponenten er ferdig tegnet i Figma. Understreken på Hover, Active og Active Hover states er laget med en inset skygge (4 opp px og med 0 px blur) slik at Figmakomponenten er enklere i oppbygning (sammenlignet med tertiærknappen) og samtidig er likere oppbygningen i kode. For å få dette til å fungere i Figma er komponenten satt til hvit bakgrunnsfarge med 0,01 % opacity. I kode er bakgrunnen helt transparent - 0% opacity slik at komponenten fungerer på andre bakgrunnsfarger.

→Note: Vi bør gjøre det samme med tertiærknappen.

Knappen er utformet i to ulike høyder: 96 px som default høyde og 64 px for Collapsed (komprimert/krympet) versjon. Collapsed versjon brukes for å redusere plassen toppmenyen tar når man scroller. På den måten får innholdet mer plass og fokus i skjermbildet. 

Komponenten er også laget for dark mode

I biblioteket ligger også en template der knappene er høyrejustert i en header. Denne er ikke publisert som en egen komponent men kan være et utgangspunkt for videre arbeid med en header/toppmeny-komponent

Figma: https://www.figma.com/file/Wh8Gfnesn25shH31sKyIaT/J%C3%B8kul-bibliotek?node-id=5098%3A0
Filmklipp av prototype: https://vimeo.com/448564195/0889a57706

**Comment by Mikkelinski** [comment]
Er denne også tenkt til å benyttes for å bygge tabs? ala https://www.carbondesignsystem.com/components/tabs/usage

Eller ser du for deg at det er kun til toppmeny?

**Comment by Steinop** [comment]
Har i utgangspunktet ikke tenkt tabs har men hvis det ikke konkurrerer for mye med hovedmenyen kan det være en mulig vei å gå. Mulig understreken skal oppføre seg annerledes hvis denne brukes som tabs (for eksempel at den animerer sidelengs til taben som er valgt?) men det må vi evt skisse mer på. Avstanden mellom tekst og strek er nok litt stor i default versjon av hovedmenyknappen.

Har vi løsninger i dag som bruker tabs? 
Hovedmenykomponenten er foreløpig ikke tegnet i kompaktversjon. Bør også se på dette hvis behovet er der.

**Comment by Mikkelinski** [comment]
Ja, er jo litt forskjeller fra tabs egentlig. Fikk tanken pga figma skissen la opp til å kunne "bygge din egen", men fremgår av navnet at er tenkt som toppmeny.

Kommer ikke på om vi har tabs i løsninger, så bør ikke forskuttere det heller

**Comment by CamillaDahlstroem** [comment]
I Flyt skal har vi en sånn hovedmeny i headeren vår - så ja, trenger kompakt versjon 🙏
Og for å svare på spørsmålet om noen bruker tabs - ja, Flyt bruker tabs et par steder. Tabene vi har laget ser ganske like ut som hovedmenyen, forskjellen er at de ligger på en slags skillestrek over innholdet.

<img width="858" alt="Skjermbilde 2020-08-18 kl  10 55 01" src="https://user-images.githubusercontent.com/11536996/90492255-47167780-e141-11ea-9938-54b42da4457c.png">

**Comment by CamillaDahlstroem** [comment]
Haha, ser nå at det faktisk er enda en annen tabmeny inni den store tabmenyen vår som ikke har en sånn skillestrek-logikk... Flyt er nok ikke stedet for å finne noen fasit på designavgjørelser 😅 Men kan være et interessant "worst case" eksempel for denne diskusjonen: Tab inni tab inni header-meny.

**Comment by Steinop** [comment]
Takk for input og bidrag @CamillaDahlstroem. Flyt er nok et eget case her siden det kjøres en slags mix av dark og light mode men send gjerne en lenke til en representativ hovedmeny som vi kan se på som utgangspunkt for kompakte menyknapper. 

@Murstam: Har dere også en kompakt header til BM?

**Comment by Murstam** [comment]
Nei, vi har ikke kompakt. Men skissene på 96px header er ikke utviklet enda. Kan hende det kan være hensiktsmessig med en kompaktversjon for oss også.

**Comment by Steinop** [comment]
Kompakt versjon:
Tok utgangspunkt i Flyt og tilpasset compact komponenten til headeren der.
Her brukes komponenten sentrert i høyden til headerfeltet i motsetning til Jøkul der headeren har samme høyde som komponenten. Må gjøre dette litt annerledes pga den mørke bakgrunnen i headeren til Flyt men komponentene vil fortsatt kunne fungere i en smalere header tilpasset komponenthøyden.

Uvisst om Flyt noensinne vil ha bruk for en collapsed versjon av headeren men da har man mulighet til å følge samme prinsipp som i desktop størrelse.

Default Compact er her 48 px høy (i en 80 px høy header) mens collapsed er 32 px (headeren forminskes til 64 px i høyden)

Film av prototype: https://vimeo.com/448913573/6b12f47bee
Figmaprototype: https://www.figma.com/proto/TkbB9ANfejDSjB2u4u1OEuqM/J%C3%B8kul-komponenter-byggesone?node-id=15876%3A121&viewport=1292%2C-375%2C0.4379739463329315&scaling=min-zoom

Figmaprototypen endrer mellom collapsed og default state på drag (Ingen hover effekter pr nå)

**Comment by piofinn** [comment]
> I Flyt skal har vi en sånn hovedmeny i headeren vår - så ja, trenger kompakt versjon 🙏
> Og for å svare på spørsmålet om noen bruker tabs - ja, Flyt bruker tabs et par steder. Tabene vi har laget ser ganske like ut som hovedmenyen, forskjellen er at de ligger på en slags skillestrek over innholdet.
> 
> <img alt="Skjermbilde 2020-08-18 kl 10 55 01" width="858" src="https://user-images.githubusercontent.com/11536996/90492255-47167780-e141-11ea-9938-54b42da4457c.png">

> Her brukes komponenten sentrert i høyden til headerfeltet i motsetning til Jøkul der headeren har samme høyde som komponenten.

Hei! Hadde ikke fått med meg alt som har skjedd her 😅 Jeg synes vi må bestemme oss for hvordan denne komponenten skal oppføre seg, altdså enten sånn som i portalen eller sånn som i Flyt. Hovedgrunnen til det er at det har veldig mye å si for hvordan komponenten fungerer (i hvert fall i kode), og sånn sett for om en kompakt versjon gir mening:

Slik det er implementert i portalen er det **ikke** menyen som setter sin egen størrelse, den har rett og slett 100% høyde av det den er inne i. Slik endrer den automatisk høyde når `Header` blir mindre, og en kompakt versjon gir slik sett ikke mening med mindre det trengs en annen skriftstørrelse/linjetykkelse e.l.

Jeg kan ikke helt se hvorfor det ikke skulle fungere i Flyt å gjøre det på samme måte. Med dagens implementasjon i kode er det for så vidt enkelt for Flyt å putte menyen inn i en container med mindre høyde for å oppnå resultatet de har i dag. Når det er sagt føler jeg at konseptet med streken under ikke passer like godt om ikke menyen fyller hele headeren. Jeg oppfatter den litt som en slags "tab", men når den ikke går helt ned til innholdet mister den sammenheng med det. Da kan man kanskje heller gå for en diamant, slik som de har i BM?

Jeg tror i hvert fall at man bør prøve å finne _ett_ konsept for disse menyene.

**Comment by Steinop** [comment]
Konseptet er egentlig det samme her men i Figma må vi ta utgangspunkt i en "standardstørrelse".
Kompaktversjonen her har tynnere strek og compact body size - dermed litt forskjellig fra desktopversjonen.

Spørsmålet her er om Flyt og @CamillaDahlstroem får det visuelle uttrykket til å fungere med hvit header?

Uansett vil ikke Flyt være førende for hvordan komponenten i kompaktversjon fungerer i forhold til designsystemets retningslinjer - den skal følge de samme prinsippene som vi bruker i hovedmenyen på Jøkul slik at fremtidige løsninger følger dette i størst mulig grad.

**Comment by onordbo** [comment]
Relevant for #2392 eller #2342? cc @wkillerud @Murstam
